### PR TITLE
Make Attribute respect suppressing comments

### PR DESCRIPTION
**Pull request by mvz**
#751

- Create new AttributeContext to handle attributes
- Make AttributeContext use the parent expression's comment

### CODE REVIEW

File: lib/reek/context/code_context.rb [hunk 1]
```diff
@@ -176,11 +176,11 @@ def tracked_visibility
       end
 
       def config
-        @config ||= if exp
-                      CodeComment.new(exp.full_comment || '').config
-                    else
-                      {}
-                    end
+        @config ||= CodeComment.new(full_comment).config
```
Comment (troessner) [inline lib/reek/context/code_context.rb]: Nice!

File: spec/reek/smells/attribute_spec.rb [hunk 2]
```diff
@@ -127,5 +127,15 @@ class OtherKlass
       '
       expect(src).to reek_of(:Attribute)
     end
+
+    it 'can be suppressed with a preceding code comment' do
```
Comment (troessner) [inline spec/reek/smells/attribute_spec.rb]: Can we make this into a "context" and add a safety test here for the dame source but without the comment? You know, "test your tests", at least when its not obvious :)

**Comment by troessner** [comment]
This PR is highly awesome. Concise and very elegant.
Ine small suggestion and the LGTM.

**Comment by troessner** [comment]
Oh, and travis ofc :laughing:

File: spec/reek/smells/attribute_spec.rb (hunk 2 above)
Comment (mvz) [inline spec/reek/smells/attribute_spec.rb]: Hm. That would basically repeat the test on line 31. Maybe I should move this test up there?

Comment (troessner) [inline spec/reek/smells/attribute_spec.rb]: >> Maybe I should move this test up there?

Good idea! Let's do that :)

**Comment by mvz** [comment]
Updated and rebased.

**Comment by troessner** [comment]
@mvz merged! Do you care for a bugfix release as well?

**Comment by troessner** [comment]
release up for review in #753

**Pull request merged**